pkg/transform/kubernetes/kubernetes.go

```diff
@@ -151,6 +152,16 @@ func (t *Transformer) TransformServices(services []object.Service) ([]runtime.Ob
 			return nil, fmt.Errorf("failed to create deployments: %s", err)
 		}
 		result = append(result, objects...)
+
+		// apply labels to all objects created by the service
+		for _, o := range result {
+			switch t := o.(type) {
+			case *api_v1.Service:
+				transformerUtil.ApplyOpenComposeLabelsToProvider(&service.Labels, &t.ObjectMeta.Labels)
+			case *ext_v1beta1.Deployment:
+				transformerUtil.ApplyOpenComposeLabelsToProvider(&service.Labels, &t.ObjectMeta.Labels)
```

Comment:
I think that adding lables to objects should be handled in functions that are responsible for creating those objects. eg `CreateServices` and `CreateDeployments`